crates/matrix-sdk-sqlite/migrations/event_cache_store/003_events.sql

```diff
@@ -0,0 +1,46 @@
+CREATE TABLE "linked_chunks" (
+    -- Identifier of the chunk, unique per room.
+    "id" INTEGER,
+    -- Which room does this chunk belong to? (hashed key shared with the two other tables)
+    "room_id" BLOB NOT NULL,
```

Comment:
Uhuh, interesting. That suggests we may have issues when there are multiple rooms, then :eyes: I'll do a few experiments and change it, if needs be — pretty sure we've been using that in other places too.

Reply:
Pfew, https://sqlite.org/datatype3.html#sort_order says that BLOB are compared with `memcmp`, so we're fine. I've added a test that used the same chunk identifier in two different rooms, and it passes, so it's all fine as is. We're using `BLOB` in many many places, so if this should've failed, it would have failed elsewhere too ^^

So now, I do buy it that using `TEXT` would offer us more, e.g. searching substrings, prefixes, etc. I don't think it's a useful use case for us, to search for a room id substring, be it in the clear or hashed.

Thanks for raising this, though! TIL :relieved: